tests/test_github.py

```diff
@@ -105,6 +130,125 @@ def test_github_retarget_pr(self) -> None:
+                .commit(branchs_first_commit_msg)
+                .commit(branchs_second_commit_msg)
+                .push(remote='origin_1')
+                .push(remote='origin_2')
+                .new_branch('feature')
+                .commit('introduce feature')
+                .push(remote='origin_1', set_upstream=False)
+                .push(remote='origin_2', set_upstream=False)
+        )
+
+        launch_command("discover", "-y")
+        expected_error_message = (
+            "Multiple non-origin remotes correspond to GitHub in this repository: origin_1, origin_2 -> aborting. \n"
+            "You can also select the repository by providing 3 git config keys: `machete.github.{remote,organization,repository}`\n"
+        )
+        with pytest.raises(MacheteException) as e:
+            launch_command("github", "retarget-pr")
+        if e:
+            assert e.value.args[0] == expected_error_message, \
+                'Verify that expected error message has appeared when given pull request to create is already created.'
+
+        # branch feature_1 present in each of the remotes, tracking data present
+        (
+            self.repo_sandbox.check_out('feature')
+                .new_branch('feature_1')
+                .commit('introduce feature 1')
+                .push(remote='origin_1')
+                .push(remote='origin_2')
+        )
+
+        launch_command("discover", "-y")
+        assert_command(
+            ['github', 'retarget-pr'],
+            'The base branch of PR #20 has been switched to `feature`\n',
+            strip_indentation=False
+        )
+
+        # branch feature_2 is not present in any of the remotes
+        (
+            self.repo_sandbox.check_out('feature')
+                .new_branch('feature_2')
+                .commit('introduce feature 2')
+        )
+
+        launch_command("discover", "-y")
+        with pytest.raises(MacheteException) as e:
+            launch_command("github", "retarget-pr")
+        if e:
+            assert e.value.args[0] == expected_error_message, \
+                'Verify that expected error message has appeared when given pull request to create is already created.'
+
+        # branch feature_2 present in only one remote: origin_1 and there is no tracking data available -> infer the remote
+        (
+            self.repo_sandbox.check_out('feature_2')
+                .push(remote='origin_1', set_upstream=False)
+        )
+
+        assert_command(
+            ['github', 'retarget-pr'],
+            'The base branch of PR #25 has been switched to `feature`\n',
+            strip_indentation=False
+        )
+
+        # branch feature_3 present in only one remote: origin_1 and has tracking data
+        (
+            self.repo_sandbox.check_out('feature_2')
+                .new_branch('feature_3')
+                .commit('introduce feature 3')
+                .push(remote='origin_1')
+        )
+
+        launch_command("discover", "-y")
```

Comment:
Hmmm discovers still left? 🤔

Reply:
Yes, it's needed when the new branch is being created, as is in line 225

Reply:
Maybe just `git machete add`? more fine-grained, easier to reason about in these tests